core/src/filesystem/image/generic.rs

```diff
@@ -51,7 +43,279 @@ impl ImageProcessor for GenericImageProcessor {
 		path: &str,
 		options: ImageProcessorOptions,
 	) -> Result<Vec<u8>, FileError> {
-		let image = Reader::open(path)?.with_guessed_format()?.decode()?;
-		Self::generate(image.as_bytes(), options)
+		let bytes = fs::read(path)?;
+		Self::generate(&bytes, options)
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use process::ImageResizeOptions;
+
+	use super::*;
+	use crate::filesystem::image::{
+		tests::{get_test_jpg_path, get_test_png_path},
+		ImageFormat, ImageProcessorOptions,
+	};
+
+	#[test]
+	fn test_generate_jpg_to_jpg() {
+		let jpg_path = get_test_jpg_path();
+		let options = ImageProcessorOptions {
+			format: ImageFormat::Jpeg,
+			..Default::default()
+		};
+
+		let buffer = GenericImageProcessor::generate_from_path(&jpg_path, options)
+			.expect("Failed to generate image buffer");
+		assert!(!buffer.is_empty());
+		// should *still* be a valid JPEG
+		assert!(
+			image::load_from_memory_with_format(&buffer, image::ImageFormat::Jpeg)
+				.is_ok()
+		);
+	}
+
+	#[test]
+	fn test_generate_jpg_to_jpg_with_rescale() {
+		let jpg_path = get_test_jpg_path();
+		let options = ImageProcessorOptions {
+			format: ImageFormat::Jpeg,
+			resize_options: Some(ImageResizeOptions::scaled(0.5, 0.5)),
+			..Default::default()
+		};
+
+		let current_dimensions =
+			image::image_dimensions(&jpg_path).expect("Failed to get dimensions");
```

Comment:
Unused variable here, was there a last part of the test that didn't get written?

Reply:
Yeah I missed cleaning it up, thanks!